### PR TITLE
Feature/mkdir p

### PR DESCRIPTION
**Pull request by szaleq**
Recursive method for creating directories.
Alternatively we could just overwrite the default `mkdir` method from WP_Filesystem to act recursively. The only difference here is the third param for native php `mkdir`.

### CODE REVIEW

**Review by jakubmikita — CHANGES_REQUESTED** [review]
Would be better to override the mkdir method for consistency

Btw. Your `$path` is wrong here. It should expect relative directory and be prefixed with `$this->path()`

File: src/Filesystem.php [hunk 1]
```diff
@@ -109,13 +109,52 @@ public function __call( $method_name, $arguments ) {
...
+
+		if ( ! $chmod ) {
+			$chmod = FS_CHMOD_DIR;
+		}
+
+		// phpcs:ignore WordPress.PHP.NoSilencedErrors.Discouraged
+		if ( ! @mkdir( $path, $chmod, true ) ) {
+			return false;
+		}
+		$this->chmod( $path, $chmod );
```
Comment (jakubmikita) [inline src/Filesystem.php]: Do you need `chmod()` method here if you are passing `$chmod` to `mkdir()`?

**Comment by jakubmikita** [comment]
Second thought - it's a method from [WP_Filesystem_Direct](https://github.com/WordPress/WordPress/blob/master/wp-admin/includes/class-wp-filesystem-direct.php) and this is the one implementation of many. What if the `WP_Filesystem_SSH2` will be used for example?

Possible edge case, but we could check if currently used filesystem is Direct and only then replace the method. If not just call the parent method.

**Comment by jakubmikita** [comment]
Tests of the new method would be highly appreciated!

**Review by jakubmikita — APPROVED** [review]
